pkg/services/extsvcauth/oauthserver/oasimpl/service.go

```diff
@@ -405,3 +406,42 @@ func (*OAuth2ServiceImpl) handleRegistrationPermissions(registration *extsvcauth
 	}
 	return selfPermissions, impersonatePermissions
 }
+
+// handlePluginStateChanged reset the client authorized grant_types according to the plugin state
+func (s *OAuth2ServiceImpl) handlePluginStateChanged(ctx context.Context, event *pluginsettings.PluginStateChangedEvent) error {
```

Comment:
Would it make sense to update a cache at the end of this function? Previously the `setClientUser` was responsible for updating it, but the caching was removed from the `setClientUser`.

Reply:
I don't understand your request sorry :pray: This function clears the cache so that we force re-fetching on next `GetExternalService` when the cache is populated back again. `setClientUser` does not update the cache anymore.

Reply:
`setClientUser`(https://github.com/grafana/grafana/pull/77248/files/d8e3236f062e49deb7f0c4e9adf3b3a23e7960f0#diff-43716239c7f3517490eb419b2e18612225345e35a574f382a9ccb9e79f690ef7R434) was responsible for updating the cache (before you removed the caching from the `setClientUser`). So in the latest version of the `handlePluginStateChanged` doesn't update the cache when it calls the `setClientUser` function and I was wondering whether this change (removed cache update) can cause troubles.